namerd/core/src/main/scala/io/buoyant/namerd/DtabStore.scala

```diff
@@ -18,6 +18,12 @@ trait DtabStore {
   def create(ns: String, dtab: Dtab): Future[Unit]
 
   /**
+   * Deletes a dtab.  Returns a DtabNamespaceDoesNotExistException if the
+   * namespace does not exist.
+   */
+  def delete(ns: String): Future[Unit]
```

Comment:
idle question only tangentially related to this review: would it be useful to have the various CRUD operations return Future[VersionedDtab] rather than Future[Unit]? That way, the caller knows the state/version of the object as of its creation/update/deletion, rather than having to call a get() later.